src/Avalonia.Base/Platform/PlatformBehavior.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Avalonia.Platform
+{
+    internal class DefaultDisposable : IDisposable
+    {
+        public void Dispose() { }
+    }
+
+    public enum PlatformBehaviorInhibitionType
+    {
+        AppSleep
+    }
+
+    public abstract class PlatformBehavior
+    {
+        public static Task<IDisposable> RequestPlatformBehaviorInhibition(PlatformBehaviorInhibitionType type, string reason)
+            => AvaloniaLocator.Current.GetService<IPlatformBehaviorInhibition>()?.RequestPlatformBehaviorInhibition(type, reason)
+            ?? Task.FromResult<IDisposable>(new DefaultDisposable());
```

Comment:
Null check here because IPlatformBehaviorInhibition is only registered in `AvaloniaNativePlatform`. Is this ok or should I right away implement it for all platforms? Also is `AvaloniaNativePlatform` actually macOS specific?